src/main/java/be/syntra/devshop/DevshopFront/controllers/ProductController.java

```diff
@@ -103,12 +104,35 @@ public String addSelectedProductFromDetailToCart(@PathVariable Long id) {
         return "redirect:/products/details/" + id;
     }
 
+    @PostMapping("/details/{id}/addreview")
+    public String addReviewToProduct(@PathVariable Long id, @ModelAttribute("review") Review review) {
+        reviewService.submitReview(id, review);
+        return "redirect:/products/details/" + id;
```

Comment:
4+ usages of String literal -> may be extracted to static final

Reply:
extracted